pkg/lib/synthesize/non_inferrable_csv.go

```diff
@@ -0,0 +1,31 @@
+package synthesize
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/operator-framework/api/pkg/lib/version"
+	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+)
+
+type RegistryV2CSV struct {
+	Name                      string                           `json:"name,omitempty"`
+	GenerateName              string                           `json:"generateName,omitempty"`
```

Comment:
If we can make a convention to generate the name from other data, I think we should. How about `${PKG_NAME}.${VERSION}`?

Reply:
Removing it is, since it provides no new information.